Repository/RepositoryFactoryCompatibility.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Doctrine\Bundle\DoctrineBundle\Repository;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\EntityRepository;
+use Doctrine\ORM\Repository\RepositoryFactory;
+use Doctrine\Persistence\ObjectRepository;
+use ReflectionMethod;
+
+if ((new ReflectionMethod(RepositoryFactory::class, 'getRepository'))->hasReturnType()) {
```

Comment:
can you please add comment indicating which branch is for which ORM version?

Reply:
Done.